polaris-core/src/main/java/org/apache/polaris/core/policy/PolicyEntity.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.policy;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.base.Preconditions;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.rest.RESTUtil;
+import org.apache.polaris.core.entity.NamespaceEntity;
+import org.apache.polaris.core.entity.PolarisBaseEntity;
+import org.apache.polaris.core.entity.PolarisEntity;
+import org.apache.polaris.core.entity.PolarisEntityType;
+
+public class PolicyEntity extends PolarisEntity {
+
+  public static final String POLICY_TYPE_CODE_KEY = "policy-type-code";
+  public static final String POLICY_DESCRIPTION_KEY = "policy-description";
+  public static final String POLICY_VERSION_KEY = "policy-version";
+  public static final String POLICY_CONTENT_KEY = "policy-content";
+
+  PolicyEntity(PolarisBaseEntity sourceEntity) {
+    super(sourceEntity);
+  }
+
+  public static PolicyEntity of(PolarisBaseEntity sourceEntity) {
+    if (sourceEntity != null) {
+      return new PolicyEntity(sourceEntity);
+    }
+
+    return null;
+  }
+
+  @JsonIgnore
+  public PolicyType getPolicyType() {
+    return PolicyType.fromCode(getPolicyTypeCode());
+  }
+
+  @JsonIgnore
+  public int getPolicyTypeCode() {
+    String policyTypeCode = getPropertiesAsMap().get(POLICY_TYPE_CODE_KEY);
+    if (policyTypeCode != null) {
+      return Integer.parseInt(policyTypeCode);
+    }
+
+    return -1;
+  }
+
+  @JsonIgnore
+  public String getDescription() {
+    return getPropertiesAsMap().get(POLICY_DESCRIPTION_KEY);
+  }
+
+  @JsonIgnore
+  public String getContent() {
+    return getPropertiesAsMap().get(POLICY_CONTENT_KEY);
+  }
+
+  @JsonIgnore
+  public String getPolicyVersion() {
+    return getPropertiesAsMap().get(POLICY_VERSION_KEY);
+  }
+
+  public static class Builder extends PolarisEntity.BaseBuilder<PolicyEntity, Builder> {
+    public Builder(Namespace namespace, String policyName) {
```

Comment:
Per [design](https://docs.google.com/document/d/1PYnCw8T1lMQfJCa1VRjk9kWMKFPtg8O0YiZ7CEoGh3Y/edit?tab=t.0#heading=h.20yx70urkehp), policy type is a required field, should we put it in the constructor as well?